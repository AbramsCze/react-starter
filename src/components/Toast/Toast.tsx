// libs
import React, { FunctionComponent } from 'react'

type OuterProps = {
  message: string;
}

export type Props = OuterProps

const Toast: FunctionComponent<Props> = (props: Props) => {
  const { message } = props

  return (
    <div className="toast">
      <div className="message">{message}</div>
    </div>
  )
}

export default Toast
