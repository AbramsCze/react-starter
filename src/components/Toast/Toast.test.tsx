// libs
import React from 'react'
import { shallow } from 'enzyme'

// others
import Toast, { Props } from './Toast'

describe('Toast component', () => {
  let props: Props

  beforeEach(() => {
    props = {
      message: 'toast message',
    }
  })

  it('should render Toast with string messages', () => {
    const renderedToast = shallow(<Toast {...props} />)
    expect(renderedToast).toMatchSnapshot()
  })
})
