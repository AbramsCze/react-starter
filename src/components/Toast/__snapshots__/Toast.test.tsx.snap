// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component should render Toast with string messages 1`] = `ShallowWrapper {}`;
